r/configure

```diff
@@ -255,8 +255,8 @@ if [ $? -eq 0 ]; then
   if arrow_built_with ARROW_S3; then
     PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_S3"
     if [ "$BUNDLED_LIBS" != "" ]; then
-      # We're depending on openssl/curl from the system, so they're not in the bundled deps
-      BUNDLED_LIBS="$BUNDLED_LIBS -lssl -lcrypto -lcurl"
+      # We're depending on openssl/curl/AWS SDK from the system, so they're not in the bundled deps
+      BUNDLED_LIBS="$BUNDLED_LIBS -lssl -lcrypto -lcurl -laws-cpp-sdk-identity-management -laws-cpp-sdk-sts -laws-cpp-sdk-cognito-identity -laws-cpp-sdk-s3 -laws-cpp-sdk-core -laws-c-event-stream -laws-checksums -laws-c-common"
```

Comment:
This seems fine, but I honestly am very confused what `BUNDLED_LIBS` is supposed to represent (or, why aren't we just adding them to `PKG_LIBS`?). Maybe it's that these are the dependencies of what is in the bundled libraries, so they need to *after* when linking?

Reply:
1. We need to add AWS SDK C++ related `-lXXX`s only when Apache Arrow C++ is built with system AWS SDK C++ (If we can require `arrow.pc`, we can implement it simply...)
2. We need to add AWS SDK C++ related `-lXXX`s before `-lssl -lcrypto -lcurl` because AWS SDK C++ depend on `-lssl -lcrypto -lcurl`

Reply:
I think we can use pkg-config to detect what libs are required from the system. In fact, we already do on L187. So is the right fix in the arrow.pc file? (Or probably in the pkg-config file that aws-sdk-cpp produces?)